util/encoding/encoding.go

```diff
@@ -444,3 +453,255 @@ func DecodeBytes(b []byte) ([]byte, []byte) {
 		b = b[i+2:]
 	}
 }
+
+// DecodeBytesDecreasing decodes a []byte value from the input buffer
+// which was encoded using EncodeBytesDecreasing. The remainder of the
+// input buffer and the decoded []byte are returned.
+func DecodeBytesDecreasing(b []byte, r []byte) ([]byte, []byte) {
+	if len(b) > 0 && b[0] == ^escape2 {
+		if len(b) == 1 {
+			panic("malformed escape")
+		}
+		if b[1] != ^escapedFF {
+			panic("unknown escape")
+		}
+		r = append(r, 0x00)
+		b = b[2:]
+	}
+
+	for {
+		i := bytes.IndexByte(b, ^escape1)
+		if i == -1 {
+			panic("did not find terminator")
+		}
+		if i+1 >= len(b) {
+			panic("malformed escape")
+		}
+
+		v := b[i+1]
+		if v == ^escapedTerm {
+			r = append(r, b[:i]...)
+			onesComplement(r)
+			return b[i+2:], r
+		}
+
+		if v == ^escapedNul {
+			r = append(r, b[:i]...)
+			r = append(r, 0xff)
+		} else {
+			panic("unknown escape")
+		}
+
+		b = b[i+2:]
+	}
+}
+
+// EncodeKey encodes values to a byte slice according to a format
+// string. Returns the byte slice containing the encoded values.
+//
+// The format string is printf-style with caveats: the primary being
+// that the "fixed" portion of the format must occur as a prefix to
+// the format. The verbs specify precisely what argument type is
+// expected and no attempt is made to perform type conversion. For
+// example, '%d' specifies an argument type of 'int64'. A panic will
+// occur if any other type is encountered.
+//
+// The verbs:
+//	%d	varint (int64) increasing
+//	%-d	varint (int64) decreasing
+//	%u	uvarint (uint64) increasing
+//	%-u	uvarint (uint64) increasing
+//	%32u	uint32 increasing
+//	%-32u	uint32 increasing
+//	%64u	uint64 increasing
+//	%-64u	uint64 increasing
+//	%s	bytes ([]byte,string) increasing
+//	%-s	bytes ([]byte,string) decreasing
+func EncodeKey(b []byte, format string, args ...interface{}) []byte {
+	i, end := 0, len(format)
+	for i < end && format[i] != '%' {
+		i++
+	}
+	b = append(b, format[:i]...)
+
+	for i < end {
+		if format[i] != '%' {
+			panic("invalid format string: " + format)
+		}
+		i++
+
+		// Process ascending flag (either "+" or "-").
+		ascending := true
+		if i < end {
+			if format[i] == '+' {
+				i++
+			} else if format[i] == '-' {
+				ascending = false
+				i++
+			}
+		}
+
+		// Process width specifier (either blank, "32" or "64").
+		width := 0
+		if i+1 < end {
+			if format[i] == '3' && format[i+1] == '2' {
+				width = 32
```

Comment:
doesn't seem like you're using the int value, this could as well be a bool and then you could set it directly.